src/sharing/components/ShareByEmail.jsx

```diff
@@ -118,14 +115,8 @@ class ShareByEmail extends React.Component {
     if (recipients.length === 0) {
       return
     }
-    this.props
-      .onSend(recipients, sharingType)
-      .then(() => {
-        this.reset()
-      })
-      .catch(() => {
-        this.reset()
-      })
+    this.props.onSend(recipients, sharingType)
+    this.reset()
```

Comment:
Since you removed the `catch` here, isn't there a chance that `this.props.onSend` results in an exception and stops the execution?

Reply:
Well seen.
If fact removing `catch` is not the only thing I removed. [`props.onSend`](https://github.com/cozy/cozy-drive/pull/653/files#diff-d6a7d92a857d281a3796807cda322a36R56) is not a promise anymore. It executes `share` from whom [I removed the return](https://github.com/cozy/cozy-drive/pull/653/files#diff-d6a7d92a857d281a3796807cda322a36R15) as it did not return anything but where used to return a empty promise that let this very function `sendSharingLink` executing `this.reset()` on `then` or on `catch`.

As reseting the input field is not related to the resolve of the promise (`then` or `catch`) and as the animation is not critical here, I simplify the code:

1. we execute `onSend` by passing `recipients` and `sharingType`
1. we execute the asynchronous fetch that creates the sharing
1. we reset the field
1. once the sharing fetch is terminated, the `ShareByEmailWrapper` will alert [an info](https://github.com/cozy/cozy-drive/pull/653/files#diff-d6a7d92a857d281a3796807cda322a36R25) or [an error](https://github.com/cozy/cozy-drive/pull/653/files#diff-d6a7d92a857d281a3796807cda322a36R31).

_Note: I will maybe merge those two components as I don't understand which one is responsible of what._